lib/config.py

```diff
@@ -77,62 +79,88 @@ class ConfigParser(object):
     Precedence is:
     cmdline > config file > argparse defaults
     """
-    def __init__(self):
-        self._CONF = self._parse_arguments()
+    def __init__(self, current_working_dir=None):
+        self.parser = argparse.ArgumentParser()
+        self._CONF = None
+        if current_working_dir:
+            self._cwd = current_working_dir
+        else:
+            self._cwd = os.getcwd()
+
+        self._setup_config_parser_options()
 
     @property
     def CONF(self):
         return self._CONF
 
-    def _parse_config(self, config_file):
+    def _setup_config_parser_options(self):
+        """
+	Configures the argument parser object to match the expected
+            configuration.
+        """
+        self.parser.add_argument('--config-file', '-c',
+                                 help='Path of the configuration file for build '
+                                     'scripts',
+                                 # NOTE(maurosr): move this to /etc in the future
+                                 default='./config.yaml')
+        self.parser.add_argument('--packages', '-p',
+                                 help='Packages to be built',
+                                 nargs='*')
+        self.parser.add_argument('--log-file', '-l',
+                                 help='Log file',
+                                 default='/var/log/host-os/builds.log')
+        self.parser.add_argument('--verbose', '-v',
+                                 help='Set the scripts to be verbose',
+                                 action='store_true')
+        self.parser.add_argument('--result-dir', '-r',
+                                 help='Directory to save the RPMs.',
+                                 default=os.path.join(self._cwd, 'result'))
```

Comment:
TBH,  I don't like the idea of our code implicitly converting paths to absolute paths in these cases.

I think it is better to allow the user to give us an absolute path is this is what they wish.

> passing an arbitrary cwd would require also making the paths got from config file relative to that cwd and I believe this disrupts the concept of "current" working dir.

Why so? If we read the paths from the configuration file, I don't see why we should apply any changes to them. The user should provide the correct paths in this case.

> Since this parameter is passed to mock, and the cwd could change along the process, using a relative path here could probably break things. os.path.abspath could be used instead.

How could the be a problem? If the CWD changes, the attribute in ConfigParser will still hold
the value we are interested in.

Maybe converting that CWD argument to an absolute path would be best.

> If you agree with the points above, the tests will need to be reworked to check absolute paths.